### PR TITLE
feat(interactive): Allow to modify filters

### PR DESCRIPTION
**Pull request by aawsome**
Adds the possibility to edit the filter used in the interactive snapshots view: 
- use 'V' to view or change the filter
- use 'Ctrl-v' to reset to the filter as given to the command

### CODE REVIEW

**Review by nardoor — CHANGES_REQUESTED** [review]
I'd advise a small change (removing the `unwrap`)

File: src/commands/tui/snapshots.rs [hunk 1]
```diff
@@ -552,6 +555,17 @@ impl<'a, P: ProgressBars, S: IndexedFull> Snapshots<'a, P, S> {
         self.get_snap_entity(|snap| snap.description.clone().unwrap_or_default())
     }
 
+    pub fn get_filter(&self) -> String {
+        toml::to_string_pretty(&self.filter).unwrap()
```
[the remark this answers is not in the document]
Comment (aawsome) [reply src/commands/tui/snapshots.rs]: is fixed.

File: src/commands/tui/snapshots.rs [hunk 2]
```diff
@@ -552,6 +555,17 @@ impl<'a, P: ProgressBars, S: IndexedFull> Snapshots<'a, P, S> {
         self.get_snap_entity(|snap| snap.description.clone().unwrap_or_default())
     }
 
+    pub fn get_filter(&self) -> Result<String> {
+        Ok(toml::to_string_pretty(&self.filter)?)
+    }
+
+    pub fn set_filter(&mut self, filter: String) {
+        if let Ok(filter) = toml::from_str::<SnapshotFilter>(&filter) {
```
Comment (nardoor) [inline src/commands/tui/snapshots.rs:563]: is there something to warn of an error in the TUI interface ?

**Review by nardoor — APPROVED** [review]
apart from a potential warning on a failed set_filters, I'd say it's good to merge

File: src/commands/tui/snapshots.rs (hunk 2 above)
Comment (aawsome) [reply src/commands/tui/snapshots.rs:563]: Let's merge it and postpone those warnings...

**Pull request merged**